frontend/src/components/core/Sidebar/SidebarNav.tsx

```diff
@@ -83,36 +85,44 @@ const SidebarNav = ({
         onMouseOver={onMouseOver}
         onMouseOut={onMouseOut}
       >
-        {appPages.map(({ pageIcon, pageName }: AppPage, pageIndex: number) => {
-          // NOTE: We use window.location to get the port instead of
-          // getBaseUriParts() because the port may differ in dev mode (since
-          // the frontend is served by the react dev server and not the
-          // streamlit server).
-          const { port, protocol } = window.location
-          const { basePath, host } = getBaseUriParts()
-
-          const portSection = port ? `:${port}` : ""
-          const basePathSection = basePath ? `${basePath}/` : ""
-
-          const navigateTo = pageIndex === 0 ? "" : pageName
-          const pageUrl = `${protocol}//${host}${portSection}/${basePathSection}${navigateTo}`
-
-          return (
-            <li key={pageName}>
-              <StyledSidebarNavLinkContainer>
-                <StyledSidebarNavLink
-                  href={pageUrl}
-                  onClick={e => {
-                    e.preventDefault()
-                    onPageChange(navigateTo)
-                  }}
-                >
-                  {pageName.replace(/_/g, " ")}
-                </StyledSidebarNavLink>
-              </StyledSidebarNavLinkContainer>
-            </li>
-          )
-        })}
+        {appPages.map(
+          ({ icon: pageIcon, pageName }: IAppPage, pageIndex: number) => {
+            pageName = pageName as string
+            // NOTE: We use window.location to get the port instead of
+            // getBaseUriParts() because the port may differ in dev mode (since
+            // the frontend is served by the react dev server and not the
+            // streamlit server).
+            const { port, protocol } = window.location
+            const baseUriParts = getBaseUriParts()
+
+            const navigateTo = pageIndex === 0 ? "" : pageName
+            let pageUrl = ""
+
+            if (baseUriParts) {
```

Comment:
In case `baseUriParts` is falsy, the `pageUrl` (=`href`) will be empty. I assume this is ok here since the navigation is handled via the `onClick` callback and not via `href`, or?

Reply:
Yep, and the callback passed to `onClick` knows to not attempt to switch pages if the frontend is disconnected from the streamlit server